akka-http-core/src/main/resources/reference.conf

```diff
@@ -278,6 +278,18 @@ akka.http {
 
       # Enable verbose debug logging for all ingoing and outgoing frames
       log-frames = false
+
+      # When there is no data transmitted in either direction, but there are active streams, send a HTTP/2 ping frame with this initial
+      # delay and subsequent interval to make sure the connection is kept alive, 0s disables sending ping frames. Must be full seconds.
+      ping-interval = 0s
+
+      # Fail the connection if a ping frame is sent and does not come back from the server within this timeout.
+      # Must be full seconds and less than the ping interval.
+      ping-timeout = 0s
+
+      # Fail the connection if more ping interchanges than this are done without any data or header frames.
+      # Setting to 0 disables this limit. Only in effect if ping-interval is enabled.
+      max-pings-without-data = 0
```

Comment:
Do we have a better understanding about why this is needed and what we want to achieve with it? Why would the user want to set this in units of "number of pings", would that correspond to a time `ping-interval * max-pings`? Maybe can move this out of this PR and discuss and work on it later.

Reply:
I can see how you wouldn't want the pings to keep a stale request stream alive _forever_, that it is pings rather than some time span was mostly because that's what the gRPC one seems to do but I agree that maybe makes more sense.

No strong opinions on my part about this being important so I'm ok with skipping it in this PR.

Reply:
Removed now